src/Patterns/templates/Scd2Snowflake.twig

```diff
@@ -0,0 +1,93 @@
+{%- else ~%}
+    {% set currentDate = '$CURRENT_DATE_TXT' %}
+    {% set infiniteDate = '9999-12-31' | quoteValue %}
+    SET CURRENT_DATE = (SELECT CONVERT_TIMEZONE('{{ timezone }}', current_timestamp()))::DATE;
+    SET CURRENT_DATE_TXT = (SELECT TO_CHAR($CURRENT_DATE, 'YYYY-MM-DD'));
+{%- endif -%}
+
+{%- if config.keepDeleteActive() -%}
+    {% set deletedEndDate = '9999-12-31 00:00:00' | quoteValue %}
+{%- else -%}
+    {% set deletedEndDate = currentDate %}
+{% endif %}
+
+-- Changed records: are not same in the snapshot and the input table --
+CREATE OR REPLACE TABLE "changed_records" AS
+    WITH "diff_records" AS (
+        SELECT {{ macro.selectFromColumns(inputColumns, "input.", true) }}
+        FROM {{ tableName.input | quoteIdentifier }} input
+            MINUS
+        SELECT {{ macro.selectFromColumns(snapshotInputColumns, "snapshot.") }}
+        FROM {{ tableName.currentSnapshot | quoteIdentifier }} snapshot
+        WHERE {{ columnName.actual | quoteIdentifier }} = 1
+    )
+    SELECT
+        {{ macro.selectFromColumns(snapshotInputColumns) }},
+        {{ currentDate }} AS {{ columnName.startDate | quoteIdentifier }},
+        {{ infiniteDate }} AS {{ columnName.endDate | quoteIdentifier }},
+        1 AS {{ columnName.actual | quoteIdentifier }},
+        0 AS {{ columnName.isDeleted | quoteIdentifier }}
+    FROM "diff_records";
+
+-- Deleted records: actual = 1 in snapshot, but missing in input table --
+CREATE OR REPLACE TABLE "deleted_records" AS
+    SELECT
+        {{ macro.selectFromColumns(snapshotInputColumns, "snapshot.") }},
+        snapshot.{{ columnName.startDate | quoteIdentifier }},
+        {{ deletedEndDate }} AS {{ columnName.endDate | quoteIdentifier }},
+        {{ deletedActualValue }} AS {{ columnName.actual | quoteIdentifier }},
+        1 AS {{ columnName.isDeleted | quoteIdentifier }}
+    FROM {{ tableName.currentSnapshot | quoteIdentifier }} snapshot
+    LEFT JOIN {{ tableName.input | quoteIdentifier }} input ON {{ snapshotInputJoinConditionSql }}
+    WHERE
+        snapshot.{{ columnName.actual | quoteIdentifier }} = 1 AND
+        input.{{ inputRandomColumn | quoteIdentifier }} IS NULL;
+
+-- Updated records: update previous version of the changed records, set actual = 0 --
+CREATE OR REPLACE TABLE "updated_records" AS
+    SELECT
+        {{ macro.selectFromColumns(snapshotInputColumns, "snapshot.") }},
+        snapshot.{{ columnName.startDate | quoteIdentifier }},
+        {{ deletedEndDate }} AS {{ columnName.endDate | quoteIdentifier }},
+        0 AS {{ columnName.actual | quoteIdentifier }},
+        0 AS {{ columnName.isDeleted | quoteIdentifier }}
+    FROM {{ tableName.currentSnapshot | quoteIdentifier }} snapshot
+    JOIN "changed_records" changed ON {{ macro.generateJoin(inputPrimaryKeyLower, "snapshot", "changed") }}
+    WHERE snapshot.{{ columnName.actual | quoteIdentifier }} = 1;
+
+{# Note: "changed_records" must be before "updated_records"  #}
```

Comment:
S tymto issue som stravil 2h+ v povodnom kode, aj v tom v Python, bolo vela chyb, ... pri niektorych situaciach - kombinaciach nastaveni - to generovalo nespravne vysledky.